args/audio/components/audio_source.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <AL/al.h>
+#include <AL/alc.h>
+#include <audio/data/audio_segment.hpp>
+
+namespace args::audio
+{
+    struct audio_source
+    {
+        friend class AudioSystem;
+    public:
+        /**
+        * @brief Function to set the pitch for the audio source
+        * @param const float pitch: new pitch value
+        */
+        void setPitch(const float pitch)
+        {
+            m_changes |= 1; // set b0
+            m_pitch = args::math::max(0.0f, pitch);
+        }
+        /**
+        * @brief Function to get the current pitch
+        */
+        float getPitch() const { return m_pitch; }
+        /**
+        * @brief Function to set the gain for the audio source
+        * @param const float gain: new gain value
+        */
+        void setGain(const float gain)
+        {
+            m_changes |= 2; // set b1
```

Comment:
a more readable way to do this is to create an enum with pitch = 1 and gain = 2 and then set m_changes |= changes::gain